### ISSUE TITLE
how to use libraries and git 

### ISSUE THREAD

**Issue by ovione** [dherges/ng-packagr#495]
## Type of Issue

```
[ ] Bug Report
[x ] Feature Request
```

## Description
This maybe doesn't belong here but because its library related I would like to know what are your tougths about it.

When building a library you may have multiple projects using this library.
What would be the git workflow used when you develop libraries with ng-packagr or any library ?

#### How To Reproduce
For example you have a lib-1.0.0 developed with agular 4.x.
This lib-1.0.0 is used by clients cl_1 and cl_2.

Now what happens when you want the same library to work with angular 5.
Then I suposse your new library will be called lib-2.0.0 developed with angular 5.x.
This lib-2.0.0 is used by clients cl_3 and cl_2.

Now my question is: What would be the best way to structure a git repository and what would be its workflow for a library like the ones like lib-1.0.0 and lib-2.0.0 ?

should lib-1.0.0 and lib-2.0.0 be in the same git repo.

master -----------------------------------------
                    \\
 branch 1   lib-1.0.0 (angular4.x)------------------------
                                                                 \\
branch 2                                                 lib-2.0.0 (angular5.x)

Here I have one repo and two branches. 
branch 1 will be for the library that uses angular 4.x and branch 2 will be for the library that uses angualr 5.x

Both libraries do the same functionality and the only difference should be the version of angular
If a bug appears in lib-1.0.0 how should it be handled ?
Should it be solved in lib-1.0.0 and then ported to lib-2.0.0 ?
We can never merge both branches to master because one is for angular 4.x and the other for angular 5.x.

This article is explaining a workflow for applications using git called git flow.
Here is the link http://datasift.github.io/gitflow/IntroducingGitFlow.html.
Basically the questions boil to how to incorporate those ideas to the libraries taken into account my prior questions ?
How do the angular guys deal with the angular itself and how do they handle versions ?

If you ahve thoughts about it please share them. I think they would be very interesting.
thanks

 

Or should we have two different repos for the same library ?

_for bugs: please include instructions how to reproduce_

_Can you reproduce the error in the integration tests in ng-packagr?_
_If possible, take a look at the [`integration/samples`](./integration/samples) and try to break one of these builds!_

_Is the error you faced in an application importing the library_
_Try to break the Angular CLI app in [`integration/consumers/ng-cli`](./integration/consumers/ng-cli)!_
